frontend/packages/console-shared/src/utils/multiselectdropdown.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+export const MultiSelectDropdown: React.FC<MultiSelectDropdownProps> = ({
+  onChange,
+  placeholder,
+  id,
+  options,
+  defaultSelected,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState<string[]>(defaultSelected || []);
+  const onSelect = (event: React.MouseEvent | React.ChangeEvent, selection: string) => {
+    let cSelected: string[] = selected;
+    if (selected.includes(selection)) {
+      cSelected = selected.filter((item) => item !== selection);
+    } else {
+      cSelected = [...selected, selection];
+    }
+    setSelected(cSelected);
+    onChange(cSelected);
+  };
+  const items: JSX.Element[] = options.map((item) => {
+    return <SelectOption key={0} value={item} />;
+  });
+  return (
+    <div>
+      <span id={id} hidden>
+        Checkbox Title
+      </span>
+
+      <Select
+        variant={SelectVariant.typeaheadMulti}
```

Comment:
I would assume most of the different variants would follow a similar logic. Maybe this would be an enhancement but it would be better to have a common component for all types of variants of Select.

Reply:
will follow up with this